src/UserManager.ts

```diff
@@ -48,32 +53,25 @@ export class UserManager extends ServiceClass {
                 email: email,
                 password: password 
         })
-        .then((response: any) => {
-            self.currentUser = response.data;
-            self.currentUser.tokenExpiration = Date.parse(response.data.expires);
-            self.currentUser.email = _email;
-            log.debug([UserManager, 'Login Success', self.currentUser]);
-            self.loadUserApplication(self.currentUser);
-            self.saveLocalUser(self.currentUser);
+        .then(res => {
+            let user = new User({
+                email: email,
+                key: res.data.key,
+                role: res.data.role,
+                token: res.data.token,
+                tokenExpiration: Date.parse(res.data.expires),
+                refresh: res.data.refresh
+            })
+            log.debug([UserManager, 'Login Success', this.currentUser]);
+            this.saveLocalUser(user);
+            return user;
         })
         .catch((err: any) => {
             throw err;
         });
     }
 
-    loadUserApplication(user: User) {
-        var _user = user;
-        this.axios().request(AuthHelper.authenticate(user, {
-            url: '/users/me/application'
-        })).then((res) => {
-            _user.application = res.data.application;
-            log.debug([UserManager, 'Loaded User Application', _user.application])
-        }).catch((err) => {
-            throw err;
-        })
-    }
-
-    createUser(email: string, password:string): Promise<any> {
+    createUser(email: string, password: string): Promise<any> {
```

Comment:
Do we want to make create user return the user from the promise to match login?

Reply:
That sounds like a good idea